File: kin-sdk/kin-sdk-lib/src/main/java/kin/sdk/models/AccountStatus.java
package kin.sdk.models;

import android.support.annotation.IntDef;

import java.lang.annotation.Retention;

import static java.lang.annotation.RetentionPolicy.SOURCE;
import static kin.sdk.models.AccountStatus.CREATED;
import static kin.sdk.models.AccountStatus.NOT_CREATED;

@Retention(SOURCE)
@IntDef({NOT_CREATED, CREATED})
public @interface AccountStatus {

    /**
     * Account was not created on blockchain network, account should be created and funded by a different account on
     * the blockchain.
     */
    int NOT_CREATED = 0;
    /**
     * Account was created, account is ready to use with kin.
     */
    int CREATED = 2;
}

File: kin-sdk/kin-sdk-lib/src/main/java/kin/sdk/models/AccountStatus.kt
package kin.sdk.models

import android.support.annotation.IntDef
import kin.sdk.models.AccountStatus.Companion.CREATED
import kin.sdk.models.AccountStatus.Companion.NOT_CREATED

@Retention(AnnotationRetention.SOURCE)
@IntDef(NOT_CREATED, CREATED)
annotation class AccountStatus {
    companion object {

        /**
         * Account was not created on blockchain network, account should be created and funded by a different account on
         * the blockchain.
         */
        const val NOT_CREATED = 0
        /**
         * Account was created, account is ready to use with kin.
         */
        const val CREATED = 2
    }
}
